crates/driver/src/tests/setup/blockchain/mod.rs

```diff
@@ -20,8 +20,62 @@ pub async fn primary_address(web3: &DynWeb3) -> ethcontract::H160 {
     web3.eth().accounts().await.unwrap()[0]
 }
 
-/// Reset the blockchain state.
-async fn reset() {
+/// An instance of geth managed by `dev-geth`. When this type is dropped, the
+/// geth instance gets shut down.
+#[derive(Debug)]
+pub struct Geth {
+    port: String,
+}
+
+impl Geth {
+    pub fn url(&self) -> String {
+        format!("http://localhost:{}", self.port)
+    }
+}
+
+impl Drop for Geth {
+    fn drop(&mut self) {
+        // Use ureq because it's blocking and doesn't rely on an async runtime.
+        ureq::delete(&format!("http://localhost:{DEV_GETH_PORT}/{}", self.port))
+            .call()
+            .unwrap();
```

Comment:
Instead of using a separate http crate just to make a blocking request, a better solution is something like
```rust
tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap().block_on(async { ... });
```
I haven't tested it, but should work.

Reply:
reqwest has blocking requests which can be enabled with a feature flag. I think we already use `reqwest` elsewhere (although - it may spin up a Tokio runtime which may be what you are trying to avoid).

Reply:
Both of your solutions lead to this:

```
thread 'tests::cases::solve::test' panicked at 'Cannot start a runtime from within a runtime. This happens because a function
 (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.', crates
/driver/src/tests/setup/blockchain/mod.rs:43:14
```

[`async` and `Drop` don' mix?](https://www.youtube.com/watch?v=sR09sACUzFU)

Reply:
Oh, right this Drop happens in async code. So it shouldn't be blocking inside of drop anyway. In that case we can do this:
```rust
tokio::task::spawn(async { issue_delete_request_through_reqwest().await; });
```
This means t drop will return before the call finishes, which is fine.

Reply:
> which is fine

This also occurred to me but I was unsure if it works - is the runtime going to wait for this task?

Reply:
From the top of my head I don't know if `#[tokio::main]` or `#[tokio::test]` wait for spawned tasks. That should be documented in the tokio docs.

Reply:
https://github.com/cowprotocol/services/pull/1121/commits/d1f6804c11466da7670f2142afceeb004a51c049